### ISSUE TITLE
如果有研究人员可以提供一个实现rt-detr热力图的代码

### ISSUE THREAD

**Issue by LiJiang97** [lyuwenyu/RT-DETR#576]
我将感激不尽。
求助，谢谢您的贡献

**Comment by wzhwantmoney** [comment]
#478 我也在想办法画热力图，我用的是pytorch版本的。我觉得这个问题中的代码很有价值，但我画出来效果不好，希望我们可以一起学习。

**Comment by zjsxlxcoming** [comment]
请问您绘制出来了么

**Comment by xsa12345** [comment]
> [#478](https://github.com/lyuwenyu/RT-DETR/issues/478) 我也在想办法画热力图，我用的是pytorch版本的。我觉得这个问题中的代码很有价值，但我画出来效果不好，希望我们可以一起学习。

请问  热力图您绘制出来了吗？

**Comment by wzhwantmoney** [comment]
> > [#478](https://github.com/lyuwenyu/RT-DETR/issues/478) 我也在想办法画热力图，我用的是pytorch版本的。我觉得这个问题中的代码很有价值，但我画出来效果不好，希望我们可以一起学习。
> 
> 请问 热力图您绘制出来了吗？

没有，但是我用特征图代替了热力图，感觉效果也还可以

**Comment by xsa12345** [comment]
我写了个热力图代码&nbsp; 效果不太行



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2025年6月25日(周三) 上午9:43
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [lyuwenyu/RT-DETR] 如果有研究人员可以提供一个实现rt-detr热力图的代码 (Issue #576)


wzhwantmoney left a comment (lyuwenyu/RT-DETR#576)
   
#478 我也在想办法画热力图，我用的是pytorch版本的。我觉得这个问题中的代码很有价值，但我画出来效果不好，希望我们可以一起学习。
  
请问 热力图您绘制出来了吗？
  
没有，但是我用特征图代替了热力图，感觉效果也还可以

**Comment by wzhwantmoney** [comment]
我的方法很笨重，就是在测试过程中把你需要的特征图保存到一个文件里，然后测试结束后对这个文件做可视化。
下面是保存特征图的代码，我是用GPT生成的，然后可视化的代码也生成一下就好了。

        # save_path = "msa_results.pt"

        # # 复制 MSA 处理前的特征
        # before_msa_0 = proj_feats[0].clone().detach()

        # # 经过 MSA 处理
        # proj_feats[0] = self.msa(proj_feats[0])
        # proj_feats[1] = self.msa(proj_feats[1])
        # proj_feats[2] = self.msa(proj_feats[2])

        # # 复制 MSA 处理后的特征
        # after_msa_0 = proj_feats[0].clone().detach()

        # # 尝试加载已有数据（如果文件存在，则追加）
        # try:
        #     existing_data = torch.load(save_path)
        #     print(f"已加载 {len(existing_data['before'])} 组数据")
        # except FileNotFoundError:
        #     existing_data = {"before": [], "after": []}
        #     print("创建新数据文件")

        # # 追加新的结果
        # existing_data["before"].append(before_msa_0.cpu())
        # existing_data["after"].append(after_msa_0.cpu())

        # # 打印当前处理的索引
        # current_index = len(existing_data["before"])
        # print(f"正在保存第 {current_index} 组 MSA 处理结果...")

        # # 保存到文件
        # torch.save(existing_data, save_path)
        # print(f"成功保存至 {save_path}，当前总数据量: {len(existing_data['before'])}")
